### PR TITLE
Schema_based_diagram

### PR DESCRIPTION
**Pull request by AJ193**
- Added schema data based on diagram
- Join tables from many-to-many relationships
- Added the FK indexes.

drawSQL-database-diagram-export-2023-08-22

### CODE REVIEW

File: Schema based on diagram.sql [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  "name" varchar,
+  PRIMARY KEY ("id")
+);
+
+CREATE TABLE "invoices" (
+  "id" int,
+  "total_amount" decimal,
+  "generated_at" timestamp,
+  "payed_at" timestamp,
+  "medical_history_id" int,
```
Comment (V-Blaze) [inline Schema based on diagram.sql]: - You have done great so far :+1: :100: But there are a few changes you need to make before moving to the next requirement, :+1: 
- [x] Looking at the diagram, `medical_history_id` is a foreign key and therefore should have a defined reference to the `Medical histories` table. and also it should be indexed. I'll suggest you update this. :+1: :+1:

File: Schema based on diagram.sql [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  PRIMARY KEY ("id")
+);
+
+CREATE TABLE "invoice_items" (
+  "id" int,
+  "unit_price" decimal,
+  "quantity" int,
+  "total_price" decimal,
+  "invoice_id" int REFERENCES "invoices"("id"), -- Specify the referenced column
+  "treatment_id" int,
```
Comment (V-Blaze) [inline Schema based on diagram.sql]: - [x] Just as I suggested above, `treatment_id` is a foreign key and it should also have reference to the `treatment` table just like`invoice_id`. It should also be indexed. I'll suggest your team update this. :+1: :+1:

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @AJ193, @Mike111222  

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


 ## Highlights
- Implemented many-many relationship correctly✔
-  Well documented code base👋 
-  Correct Git flow✔

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ShadyShawkat — APPROVED** [review]
## Hi @AJ193, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 
- Your readme looks professional ✔️ 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**